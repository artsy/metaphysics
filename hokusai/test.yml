version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
