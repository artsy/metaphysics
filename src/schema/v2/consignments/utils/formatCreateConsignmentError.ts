import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
