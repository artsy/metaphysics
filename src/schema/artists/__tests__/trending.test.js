import gql from "test/gql"
import { runQuery } from "test/utils"

describe("Trending Artists", () => {
  it("makes a call for trending artists", async () => {
    const query = gql`
      {
        trending_artists(name: ARTIST_FAIR) {
          artists {
            id
          }
        }
      }
    `
    const rootValue = {
      deltaLoader: ({ name }) =>
        name === "artist_fair" && Promise.resolve({ ortina: null, xtina: null, cached: null, context_type: null }),
      artistLoader: id => Promise.resolve({ id, birthday: "1900", artworks_count: 100 }),
    }

    const { trending_artists: { artists } } = await runQuery(query, rootValue)
    expect(artists).toMatchSnapshot()
  })
})
