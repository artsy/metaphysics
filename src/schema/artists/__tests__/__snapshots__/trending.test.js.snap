// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trending Artists makes a call for trending artists 1`] = `
Array [
  Object {
    "id": "ortina",
  },
  Object {
    "id": "xtina",
  },
]
`;
